core/src/org/javarosa/core/model/instance/TreeElement.java

```diff
@@ -936,56 +931,16 @@ public void populate(TreeElement incoming, FormDef f) {
                 this.setValue(AnswerDataFactory.templateByDataType(this.dataType).cast(value.uncast()));
             }
         } else {
-            Vector names = new Vector();
-
-            // remove all default repetitions from skeleton data model (_preserving_ templates, though)
+            // remove all default repetitions from skeleton data model, preserving templates
             for (int i = 0; i < this.getNumChildren(); i++) {
                 TreeElement child = this.getChildAt(i);
-                if (child.getMaskVar(MASK_REPEATABLE) && child.getMult() != TreeReference.INDEX_TEMPLATE) {
+                if (child.getMaskVar(MASK_REPEATABLE) &&
+                        child.getMult() != TreeReference.INDEX_TEMPLATE) {
                     this.removeChildAt(i);
                     i--;
                 }
             }
 
-            //We used to wipe out non-repeated elements, but we won't anymore
-            //since we commonly use the pattern.
-            for (int i = 0; i < this.getNumChildren(); i++) {
-                TreeElement child = this.getChildAt(i);
-                names.addElement(child.getName());
-            }
-
-            // make sure ordering is preserved (needed for compliance with xsd schema)
-            if (this.getNumChildren() != names.size()) {
-                throw new RuntimeException("sanity check failed");
-            }
-
-            for (int i = 0; i < this.getNumChildren(); i++) {
-                TreeElement child = this.getChildAt(i);
-                String expectedName = (String)names.elementAt(i);
-
-                if (!child.getName().equals(expectedName)) {
```

Comment:
The test expression in this conditional is never true!
Hence, remove the conditional altogether.